tests/mnle_test.py

```diff
@@ -0,0 +1,199 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
+
+import pytest
+import torch
+from pyro.distributions import InverseGamma
+from torch.distributions import Beta, Binomial, Gamma
+
+from sbi.inference import MNLE, MCMCPosterior, likelihood_estimator_based_potential
+from sbi.inference.potentials.base_potential import BasePotential
+from sbi.utils import BoxUniform, likelihood_nn, mcmc_transform
+from sbi.utils.torchutils import atleast_2d
+from sbi.utils.user_input_checks_utils import MultipleIndependent
+from tests.test_utils import check_c2st
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize("device", ("cpu", "cuda"))
+def test_mnle_on_device(device):
+
+    # Generate mixed data.
+    num_simulations = 100
+    theta = torch.rand(num_simulations, 2)
+    x = torch.cat(
+        (torch.rand(num_simulations, 1), torch.randint(0, 2, (num_simulations, 1))),
+        dim=1,
+    )
+
+    # Train and infer.
+    prior = BoxUniform(torch.zeros(2), torch.ones(2))
+    trainer = MNLE(prior=prior, device=device)
+    trainer.append_simulations(theta, x).train(max_num_epochs=1)
+
+    # Test sampling on device.
+    posterior = (
+        trainer.append_simulations(theta, x).train(max_num_epochs=1).build_posterior()
+    )
+    posterior.sample((1,), x=x[0], show_progress_bars=False)
+
+
+@pytest.mark.parametrize("sampler", ("mcmc", "rejection", "vi"))
+def test_mnle_api(sampler):
+
+    # Generate mixed data.
+    num_simulations = 100
+    theta = torch.rand(num_simulations, 2)
+    x = torch.cat(
+        (torch.rand(num_simulations, 1), torch.randint(0, 2, (num_simulations, 1))),
+        dim=1,
+    )
+
+    # Train and infer.
+    prior = BoxUniform(torch.zeros(2), torch.ones(2))
+    x_o = x[0]
+    # Build estimator manually.
+    density_estimator = likelihood_nn(model="mnle", **dict(tail_bound=2.0))
+    trainer = MNLE(density_estimator=density_estimator)
+    mnle = trainer.append_simulations(theta, x).train(max_num_epochs=1)
+
+    # Test different samplers.
+    posterior = trainer.build_posterior(prior=prior, sample_with=sampler)
+    posterior.set_default_x(x_o)
+    if sampler == "vi":
+        posterior.train()
+    posterior.sample((1,), show_progress_bars=False)
+
+    # MNLE should work with the default potential as well.
+    potential_fn, parameter_transform = likelihood_estimator_based_potential(
+        mnle, prior, x_o
+    )
+    posterior = MCMCPosterior(
+        potential_fn,
+        proposal=prior,
+        theta_transform=parameter_transform,
+        init_strategy="proposal",
+    )
+    posterior.sample((1,), show_progress_bars=False)
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    "sampler",
+    (
+        "mcmc",
+        "rejection",
+        # "vi",  # Failing because of transformed space dimension mismatch.
```

Comment:
why is this failing? Is this worth creating an issue?

Reply:
yes, I will create one.